hcloud/client_handler_debug.go

```diff
@@ -0,0 +1,54 @@
+package hcloud
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httputil"
+)
+
+func wrapDebugHandler(wrapped handler, output io.Writer) handler {
+	return &debugHandler{wrapped, output}
+}
+
+type debugHandler struct {
+	handler handler
+	output  io.Writer
+}
+
+func (h *debugHandler) Do(req *http.Request, v any) (resp *Response, err error) {
+	// Clone the request, so we can redact the auth header, read the body
+	// and use a new context.
+	cloned := req.Clone(context.Background())
+
+	if req.ContentLength > 0 {
+		cloned.Body, err = req.GetBody()
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This code to clone the request is is duplicated in `retryHandler.Do()`.

Reply:
Not exactly, in the retry handler we reuse the request context, for the debug dump, we create a new context.

I'll refactor this.